backend/src/bundles/users/user.service.ts

```diff
@@ -49,10 +53,48 @@ class UserService implements Service {
     }
 
     public async update(
+        query: Record<string, number>,
+        payload: UserUpdateProfileRequestDto,
+    ): Promise<UserAuthResponseDto | null> {
+        try {
+            const userId = query['id'] as number;
+            const existingUser = await this.userRepository.find({
+                id: userId,
+            });
```

Comment:
Already checked in plugin and repository, that's a lot of unnecessary db queries.